api/src/backend/api/v1/serializers.py

```diff
@@ -2231,3 +2240,109 @@ def update(self, instance, validated_data):
             IntegrationProviderRelationship.objects.bulk_create(new_relationships)
 
         return super().update(instance, validated_data)
+
+
+class LighthouseConfigSerializer(RLSSerializer):
+    """
+    Serializer for the LighthouseConfig model.
+    """
+
+    api_key = serializers.CharField(write_only=True, required=False)
+
+    class Meta:
+        model = LighthouseConfig
+        fields = [
+            "id",
+            "name",
+            "api_key",
+            "model",
+            "temperature",
+            "max_tokens",
+            "business_context",
+            "is_active",
+            "inserted_at",
+            "updated_at",
+            "url",
+        ]
+        extra_kwargs = {
+            "id": {"read_only": True},
+            "inserted_at": {"read_only": True},
+            "updated_at": {"read_only": True},
+        }
+
+    def to_representation(self, instance):
+        ret = super().to_representation(instance)
+        # Add back the masked API key for display
+        ret["api_key"] = "*" * len(instance.api_key) if instance.api_key else None
+        return ret
+
+
+class LighthouseConfigCreateSerializer(RLSSerializer, BaseWriteSerializer):
+    """Serializer for creating new AI configurations."""
+
+    api_key = serializers.CharField(write_only=True, required=True)
+
+    class Meta:
+        model = LighthouseConfig
+        fields = [
+            "name",
+            "api_key",
+            "model",
+            "temperature",
+            "max_tokens",
+            "business_context",
+            "is_active",
+        ]
+
+    def validate(self, attrs):
+        tenant_id = self.context.get("request").tenant_id
+        if LighthouseConfig.objects.filter(tenant_id=tenant_id).exists():
+            raise serializers.ValidationError(
+                {
+                    "tenant_id": "AI configuration already exists for this tenant. Use PUT to update."
+                }
+            )
+        return super().validate(attrs)
```

Comment:
You are still missing what I suggested in https://github.com/prowler-cloud/prowler/pull/7508#discussion_r2052211175.

Reply:
Not changing this as discussed on Slack thread - https://prowler-workspace.slack.com/archives/C08BGLVDJ4D/p1745935957195789